### ISSUE TITLE
ModuleNotFoundError: No module named 'NeuralNRT'

### ISSUE THREAD

**Issue by erezposner** [DeformableFriends/NeuralTracking#3]
Hi, I have followed the installation instructions and running example_viz.py I get this error:
```
Traceback (most recent call last):
  File "example_viz.py", line 7, in <module>
    from utils import image_proc
  File "/home/user/Repositories/NeuralTracking/utils/image_proc.py", line 6, in <module>
    from NeuralNRT._C import backproject_depth_ushort as backproject_depth_ushort_c
ModuleNotFoundError: No module named 'NeuralNRT'
```
I tried to see if there were any problems during installation of NeuralNRT but I get this verbose (which I think is allright).
```
running bdist_egg
running egg_info
writing NeuralNRT.egg-info/PKG-INFO
writing dependency_links to NeuralNRT.egg-info/dependency_links.txt
writing top-level names to NeuralNRT.egg-info/top_level.txt
reading manifest file 'NeuralNRT.egg-info/SOURCES.txt'
writing manifest file 'NeuralNRT.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_ext
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/NeuralNRT
copying build/lib.linux-x86_64-3.6/NeuralNRT/_C.cpython-36m-x86_64-linux-gnu.so -> build/bdist.linux-x86_64/egg/NeuralNRT
warning: install_lib: byte-compiling is disabled, skipping.

creating stub loader for NeuralNRT/_C.cpython-36m-x86_64-linux-gnu.so
warning: install_lib: byte-compiling is disabled, skipping.

creating build/bdist.linux-x86_64/egg/EGG-INFO
copying NeuralNRT.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying NeuralNRT.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying NeuralNRT.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying NeuralNRT.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
writing build/bdist.linux-x86_64/egg/EGG-INFO/native_libs.txt
zip_safe flag not set; analyzing archive contents...
creating 'dist/NeuralNRT-1.0-py3.6-linux-x86_64.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing NeuralNRT-1.0-py3.6-linux-x86_64.egg
Removing /home/user/anaconda3/envs/nnrt/lib/python3.6/site-packages/NeuralNRT-1.0-py3.6-linux-x86_64.egg
Copying NeuralNRT-1.0-py3.6-linux-x86_64.egg to /home/user/anaconda3/envs/nnrt/lib/python3.6/site-packages
NeuralNRT 1.0 is already the active version in easy-install.pth

Installed /home/user/anaconda3/envs/nnrt/lib/python3.6/site-packages/NeuralNRT-1.0-py3.6-linux-x86_64.egg
Processing dependencies for NeuralNRT==1.0
Finished processing dependencies for NeuralNRT==1.0
```

Thank you

**Comment by pablopalafox** [comment]
Hey @erezposner, I just tried installing from scratch and it worked for me. I did notice that you are using python 3.6. Could you try with python 3.7? In principle that should have been taken care of when installing the conda environment in the repo, but maybe double check that after running `conda env create --file resources/env.yml` you do get version 3.7 or higher.

**Comment by erezposner** [comment]
For anyone:

I've used python 3.6 because open3d is problematic to install on ubuntu 16.04 (needed to build from source and other issues) and so, that was my initial approach.
I also use RTX3080 so there are discrepancies between pytorch1.4 and my HW architecture.

I've found the best solution was to use nvidia-docker with x11 forwarding.
I will contribute to this repo in the next couple of days and add a dockerfile.

**Comment by joos2010kj** [comment]
Hello -- I'm getting the same issue as yours, so I was wondering how you resolved the issue.

**Comment by erezposner** [comment]
@joos2010kj See my previous comment. 
I've contributed a dockerfile and a script to build it to this repo.

Using this approach would alleviate any library discrepancies and installation issues that you might experience.

**Comment by joos2010kj** [comment]
Thanks for the contribution.

Have you had this issue while running by the way?:

> Step 19/19 : WORKDIR /workspace
>  ---> Running in e2b60ba4e024
> Removing intermediate container e2b60ba4e024
>  ---> 9b18f6a7cfd2
> [Warning] One or more build-args [NNRT_PATH] were not consumed
> Successfully built 9b18f6a7cfd2
> Successfully tagged ml_docker_ne:latest
> portainer_dat
> Error response from daemon: No such container: portainer
> 114e062d8b8dec21b8cf8c5b7091f69d5d5bbbbef04e841dda458b9f1185f6da
> docker: Error response from daemon: invalid mode: /workspace/local_station.
> See 'docker run --help'.
> X@X-PC:/mnt/c/Users/X/Desktop/repo/NeuralTracking$

I used WSL2 (Ubuntu) on Windows 10 with a docker running.

**Comment by erezposner** [comment]
I am not sure but have you tried changing the LOCAL_SRC_DIR in the `start_nnrt.sh` script?

**Comment by joos2010kj** [comment]
Yea, it is set to be: LOCAL_SRC_DIR=C:\\\\Users\\\\X\\\\Desktop\\\\repo\\\\NeuralTracking-Docker

and I ran this to remove some carriages: perl -pi -e 's/\r//' start_nnrt.sh

I started over from scratch with a new git clone, and I got the same result.

> X@X-PC:/mnt/c/Users/X/Desktop/repo/NeuralTracking-Docker$ sh start_nnrt.sh
> xhost:  unable to open display ""
> portainer
> Error response from daemon: No such container: ml_docker_ne
> portainer
> Error: No such container: ml_docker_ne
> Sending build context to Docker daemon  179.3MB
> Step 1/19 : FROM nvcr.io/nvidia/pytorch:20.03-py3 AS main
>  ---> 16c4987611fa
> Step 2/19 : RUN apt-get update
>  ---> Using cache
>  ---> c7cc410c1fd5
> Step 3/19 : RUN apt-get install -y xorg-dev             libglu1-mesa-dev                python3-dev             libsdl2-dev             libc++-7-dev            libc++abi-7-dev                 ninja-build             libxi-dev         libtbb-dev            libosmesa6-dev         libusb-1.0-0-dev                 build-essential                 manpages-dev
>  ---> Using cache
>  ---> 05c88c91fb74
> Step 4/19 : COPY --from=nvidia/opengl:1.0-glvnd-runtime-ubuntu16.04   /usr/local/lib/x86_64-linux-gnu   /usr/local/lib/x86_64-linux-gnu
>  ---> Using cache
>  ---> e0485b6f898c
> Step 5/19 : COPY --from=nvidia/opengl:1.0-glvnd-runtime-ubuntu16.04   /usr/local/share/glvnd/egl_vendor.d/10_nvidia.json   /usr/local/share/glvnd/egl_vendor.d/10_nvidia.json
>  ---> Using cache
>  ---> a695bab81eb3
> Step 6/19 : RUN echo '/usr/local/lib/x86_64-linux-gnu' >> /etc/ld.so.conf.d/glvnd.conf &&     ldconfig &&     echo '/usr/local/$LIB/libGL.so.1' >> /etc/ld.so.preload &&     echo '/usr/local/$LIB/libEGL.so.1' >> /etc/ld.so.preload
>  ---> Using cache
>  ---> ce6c2c661e38
> Step 7/19 : ENV NVIDIA_VISIBLE_DEVICES     ${NVIDIA_VISIBLE_DEVICES:-all}
>  ---> Using cache
>  ---> 2807d26d4656
> Step 8/19 : ENV NVIDIA_DRIVER_CAPABILITIES     ${NVIDIA_DRIVER_CAPABILITIES:+$NVIDIA_DRIVER_CAPABILITIES,}graphics
>  ---> Using cache
>  ---> 0696954d72c9
> Step 9/19 : COPY docker_req/req.txt /workspace/reqs/
>  ---> Using cache
>  ---> 1af2e46bc676
> Step 10/19 : RUN pip install -r /workspace/reqs/req.txt
>  ---> Using cache
>  ---> 24bcbda4405b
> Step 11/19 : RUN pip install torch==1.7.0+cu110 torchvision==0.8.1+cu110 torchaudio===0.7.0 -f https://download.pytorch.org/whl/torch_stable.html
>  ---> Using cache
>  ---> 9d6b8ab23d2c
> Step 12/19 : COPY . /workspace/NNRT/
>  ---> Using cache
>  ---> 5cae621eac07
> Step 13/19 : WORKDIR /workspace/NNRT/csrc
>  ---> Using cache
>  ---> e1fba2e38f0c
> Step 14/19 : RUN python setup.py install
>  ---> Using cache
>  ---> 406f282e3546
> Step 15/19 : RUN mkdir /workspace/local_station
>  ---> Using cache
>  ---> af34a13de00b
> Step 16/19 : EXPOSE 8097
>  ---> Using cache
>  ---> bcd6cd79541c
> Step 17/19 : EXPOSE 6006
>  ---> Using cache
>  ---> f6a4e1297edf
> Step 18/19 : EXPOSE 8888
>  ---> Using cache
>  ---> 232f0f733104
> Step 19/19 : WORKDIR /workspace
>  ---> Using cache
>  ---> a90247e7e4e1
> [Warning] One or more build-args [NNRT_PATH] were not consumed
> Successfully built a90247e7e4e1
> Successfully tagged ml_docker_ne:latest
> portainer_dat
> Error response from daemon: No such container: portainer
> 961ba6929c8058eb7186e581a51b10529150ee25ab3c809cde482d8589569f59
> docker: Error response from daemon: invalid mode: /workspace/local_station.
> See 'docker run --help'.
> X@X-PC:/mnt/c/Users/X/Desktop/repo/NeuralTracking-Docker$

**Comment by erezposner** [comment]
It seems that Docker Toolbox for mapping a volume in windows is a bit different than in ubuntu (which this docker was build and tested on).
I would suggest exploring google for a solution. Maybe this would help to start:
https://stackoverflow.com/questions/50540721/docker-toolbox-error-response-from-daemon-invalid-mode-root-docker 
good luck

**Comment by joos2010kj** [comment]
Yup.  I tested with Linux dualboot and it worked.

For those who are having difficulties, here's a note:
- Docker approach is (i think) not tested with Windows (nor WSL).  I tried it using my macbook too but it didn't work (I think the culprit is GPU).  So use docker only if you are using Linux OS.
- Make sure to have nvidia driver installed (check with nvidia-smi) and nvidia-cuda-toolkit installed.  I didnt have them installed at first, but installing fixed issues.  


Tested with...
OS: Ubuntu 20.04
GPU: 2070S
Nvidia Driver Version: 450
Approach: Docker

Thanks @erezposner
